plugin/grails-app/controllers/grails/plugin/springsecurity/LoginController.groovy

```diff
@@ -104,18 +105,19 @@ class LoginController {
         String msg = ''
         def exception = session[WebAttributes.AUTHENTICATION_EXCEPTION]
         if (exception) {
+            def locale = GrailsWebRequest.lookup().getLocale() ?: Locale.getDefault()

```

Comment:
@matrei  "should" ?  I took the logic from [FormatTagLib.groovy#L336-L349](https://github.com/grails/grails-gsp/blob/7ad985676e6217ad331f75aafad8258c1cb9d8cb/grails-plugin-gsp/src/main/groovy/org/grails/plugins/web/taglib/FormatTagLib.groovy#L336-L349)
can you confirm 100% it will give the same result.  If so,  FormatTagLib would need to be updated as well.

Reply:
"Should" as in I have not tested it in this context.

According to https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/i18n/LocaleContextHolder.html#getLocale():
> Return the Locale associated with the current thread, if any, or the system default Locale otherwise

From https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-servlet/localeresolver.html:
> When a request comes in, the DispatcherServlet looks for a locale resolver and, if it finds one, it tries to use it to set the locale.

By default in a Grails app, that is a `SessionLocaleResolver`, set by `GrailsI18nPlugin.doWithSpring()`.

`GrailsWebRequest.lookup().getLocale()` will eventually end up using the same `LocaleResolver` to get the `Locale` but it will not fallback to using the system `Locale`.